specs/docs/merkle-tree/merkle-tree.md

```diff
@@ -0,0 +1,84 @@
+# Merkle Tree
+
+To keep our merkle tree verification succinct, efficient and friendly with [SnarkyJS](https://github.com/o1-labs/snarkyjs). A poseidon merkle tree will be used to prove the immutability of the data. The **Sparse Merkle Tree** is utilized as an adaptation of the conventional Merkle Tree.
+
+## Sparse Merkle Tree (SMT)
+A Sparse Merkle Tree (SMT) is a variant of a standard Merkle tree that is optimized for scenarios where the data set is very large, but only a small portion of it is populated with values. You could refer to the following article to learn more: [What’s a Sparse Merkle Tree?](https://medium.com/@kelvinfichter/whats-a-sparse-merkle-tree-acda70aeb837).
+
+### Advantages of SMT
+Sparse Merkle Trees (SMTs) offer several benefits:
+
+1) **Efficiency**: They allow efficient storage of large, sparse datasets with minimal memory overhead.
+2) **Security**: Like traditional Merkle trees, they provide tamper-proof data structure and enable cryptographic verification of data integrity.
+3) **Proof Size**: The proof size for SMTs is consistent, regardless of the tree's size, making them optimal for scenarios where frequent proofs are required.
+4) **Flexible Updating**: They support efficient updates and insertions even in massive datasets.
+
+### Ways to store Merkle Tree on IPFS
+Here are different ways you could store a Merkle Tree on IPFS:
+
+1) **JSON Serialization**: One of the simplest ways to store a Merkle Tree in IPFS is to convert the Merkle Tree to a JSON structure and then save that to IPFS. This is a straightforward method but can be inefficient for large trees, as the entire tree needs to be retrieved even if you're only interested in certain parts of it.
```

Comment:
Could we consider BSON and define a structure to store Merkle tree?

Reply:
Done